src/modalities/evaluator.py

```diff
@@ -120,6 +77,51 @@ def evaluate(
         # )
         return result_dict
 
+    def _process_batches_in_data_loader(
+        self,
+        model: NNModel,
+        data_loader: LLMDataLoader,
+        global_train_sample_id: int,
+        local_sample_id_to_global_sample_id: Callable[[int], int],
+    ) -> Tuple[Dict[str, torch.Tensor], Dict[str, torch.Tensor]]:
+        losses = [f.create(self.local_rank) for f in self._loss_factories]
+        metrics = [f.create(self.local_rank) for f in self._metric_factories]
+
+        for batch_id, batch in enumerate(data_loader):
+            self.evaluate_batch(
+                batch=batch,
+                model=model,
+                loss_functions=losses,
+                metrics=metrics,
+            )
+
+            local_dataset_sample_id = Evaluator._get_local_sample_id(
+                batch_id=batch_id, batch_size=data_loader.batch_size
+            )
+            global_dataset_sample_id = local_sample_id_to_global_sample_id(local_dataset_sample_id)
+            Evaluator._publish_progress(
+                batch_progress_publisher=self.batch_progress_publisher,
+                global_train_sample_id=global_train_sample_id,
+                global_dataset_sample_id=global_dataset_sample_id,
+                dataloader_tag=data_loader.dataloader_tag,
+            )
+
+        return {loss: loss.compute() for loss in losses}, {metric: metric.compute() for metric in metrics}
```

Comment:
maybe it makes sense to simplify losses and metrics just as measures? Maybe we can find a better word even?

Reply:
Also, what about replacing `compute` with `aggregate`?